### PR TITLE
Fixed #1687 - Removed Growth Experiment 3 Code

### LINKED ISSUE #1687: Disable/remove Growth Experiment 3

### PR DESCRIPTION
**Pull request by maxxcrawford**
Note – I'm leaving some remnants of experiment code for future experiment usage. I'll comment on specific code that should be retained.

### CODE REVIEW

File: controllers/user.js [hunk 1]
```diff
@@ -568,6 +550,7 @@ async function getBreachStats(req, res) {
 
 
 function logout(req, res) {
+  // Growth Experiment
   if (AppConstants.EXPERIMENT_ACTIVE) {
```
Comment (maxxcrawford) [inline controllers/user.js:554]: Unreachable Code – This is gated on the ENV variable, `EXPERIMENT_ACTIVE` set to 1.

File: controllers/utils.js [hunk 2]
```diff
@@ -33,13 +33,32 @@ function hasUserSignedUpForRelay(user) {
...
+  }
+
+  // If the user doesn't have an English variant langauge selected as their primary language,
+  // we do not enroll them in the experiment.
+  const lang = req.headers["accept-language"].split(",");
+  if (language && !lang[0].includes(language)) {
+    log.debug("Preferred language is not English variant: ", lang[0]);
+    return false;
+  }
+
```
Comment (maxxcrawford) [inline controllers/utils.js:61]: Leaving this function, but not exposing it to any views.

File: public/js/fxa-analytics.js [hunk 3]
```diff
@@ -155,6 +155,7 @@ function setGAListeners(){
       });
     });
 
+    // Growth Experiment
     if (document.body.dataset.experiment) {
```
Comment (maxxcrawford) [inline public/js/fxa-analytics.js:159]: Unreachable Code – This is gated on the ENV variable, `EXPERIMENT_ACTIVE` set to 1.

File: public/js/fxa-analytics.js [hunk 4]
```diff
@@ -211,6 +212,8 @@ function setGAListeners(){
     ga("create", "UA-77033033-16");
     ga("set", "anonymizeIp", true);
     ga("set", "dimension6", `${document.body.dataset.signedInUser}`);
+
+    // Growth Experiment
     if (document.body.dataset.experiment) {
```
Comment (maxxcrawford) [inline public/js/fxa-analytics.js:217]: Unreachable Code – This is gated on the ENV variable, `EXPERIMENT_ACTIVE` set to 1.

File: tests/controllers/home.test.js [hunk 5]
```diff
@@ -2,20 +2,10 @@
 
 const AppConstants = require("../../app-constants");
 const home = require("../../controllers/home");
-const { getExperimentBranch } = require("../../controllers/utils");
```
Comment (maxxcrawford) [inline tests/controllers/home.test.js:5]: Removing the tests for the function as its' not longer included in the home controller.

File: views/scan.hbs [hunk 6]
```diff
@@ -15,15 +15,7 @@
...
-          class="open-oauth ga-growth-ping {{#if experimentBranchB}} temp-marketing-btn-blue experiment {{else}} blue-link temp-marketing-alert-me-link {{/if}}"
-          {{#if experimentBranchB}}
-            {{> analytics/fxa id="fx-monitor-alert-me-blue-btn-top" }}
-          {{else}}
-            {{> analytics/fxa id="fx-monitor-alert-me-blue-link" }}
-          {{/if}}
-          data-event-category="Alert me about new breaches">
-            {{ getString "alert-about-new-breaches" }}
-          </button>
+          class="open-oauth blue-link temp-marketing-alert-me-link" {{> analytics/fxa id="fx-monitor-alert-me-blue-link" }} data-event-category="Alert me about new breaches"> {{ getString "alert-about-new-breaches" }} </button>
```
Comment (maxxcrawford) [inline views/scan.hbs:18]: This is the primary patch that needed to be reverted.

**Comment by maxxcrawford** [comment]
See this PR up on this review site: https://monitor-v2.herokuapp.com/

Note – The `EXPERIMENT_ACTIVE` ENV variable has been set to `0`.

**Comment by maxxcrawford** [comment]
See this PR up on this review site: https://monitor-v2.herokuapp.com/

Note – The `EXPERIMENT_ACTIVE` ENV variable has been set to `0`

File: controllers/user.js (hunk 1 above)
Comment (groovecoder) [reply controllers/user.js:554]: nit: this doesn't use a type-safe value check like above? `AppConstants.EXPERIMENT_ACTIVE === "1"` So `EXPERIMENT_ACTIVE=false` would evaluate to `true` here. (❤️ Javascript)

File: tests/controllers/home.test.js (hunk 5 above)
Comment (groovecoder) [reply tests/controllers/home.test.js:5]: Is there any harm in leaving the tests in place to make sure we don't break the function? We might use it in the future, so would be nice to make sure we don't break it between now and then.

File: views/scan.hbs (hunk 6 above)
Comment (groovecoder) [reply views/scan.hbs:18]: @lesleyjanenorton - look good?

**Comment by groovecoder** [comment]
I spot-checked scan, sign up, sign in, add email address, remove email address. All work and look good.

File: views/scan.hbs (hunk 6 above)
Comment (lesleyjanenorton) [reply views/scan.hbs:18]: looks good and runs fine locally

File: controllers/user.js (hunk 1 above)
Comment (maxxcrawford) [reply controllers/user.js:554]: This is actually a bug from the experiment. Adding back the logic for `EXPERIMENTS_ENABLED` to gate it correctly.

File: tests/controllers/home.test.js (hunk 5 above)
Comment (maxxcrawford) [reply tests/controllers/home.test.js]: Added tests back, and included a language-specific one that should fail in the requested language is not in the first/primary position of the `accept-language` headers.

File: controllers/utils.js [hunk 7]
```diff
@@ -33,10 +33,11 @@ function hasUserSignedUpForRelay(user) {
   return false;
 }
 
-function getExperimentBranch(req, language, sorterNum) {
+function getExperimentBranch(req, sorterNum = false, language = false) {
```
Comment (groovecoder) [inline controllers/utils.js:36]: This also reverses the order of the parameters? I don't see the calls to the function changed to match?

Comment (maxxcrawford) [reply controllers/utils.js:36]: It was commented out/broken previously. I had added back in the language logic for future use. It's now working. For the current experiment, there was no language filtering, so no args is okay.

**Pull request merged**